src/instruction-search.njk

```diff
@@ -0,0 +1,50 @@
+---
+title: Instruction Search
+permalink: search/index.html
+---
+<h1>Instruction Search</h1>
+
+<p>Search by mnemonic or long name. Hit Enter on an exact mnemonic match to jump straight to the instruction details.</p>
+
+<input id="search" placeholder="Search mnemonic, e.g. addi" style="padding:8px; width: 100%; max-width: 480px;" />
+
+<ul id="list">
+{% for inst in instructions %}
+  <li data-name="{{ inst.name }} {{ inst.longName | lower }}">
+    <a href="/instructions/{{ inst.name }}/">{{ inst.name }}</a>
+  </li>
+{% endfor %}
+</ul>
+
+<script>
```

Comment:
We can do this in a subsequent PR, but I think it would be a really good idea to move our client-side JavaScript code to a separate JS file instead of inlining it in the HTML. (Again, let's do that in the next PR!)